scripts/CameraJoint.gd

```diff
@@ -12,15 +12,28 @@ func _ready():
 
 func _physics_process(delta):
 	# zoom the camera in or out based on the scroll wheel
-	if Input.is_action_just_released("scroll_up"):
+	if Input.is_action_just_released("scroll_up") and spring_length > 0:
 		spring_length -= scroll_speed * delta
```

Comment:
Is it possible that `spring_length` goes below 0 during this operation, and does it matter?

What about:
```suggestion
		spring_length -= min(scroll_speed * delta, spring_length)
```

Reply:
On line 23, there's `spring_length = clamp(spring_length, 0, 7)`, which ensures the spring_length is between 0 and 7.

Reply:
I saw that, but before that happens you might have reduced/increased beyond that